pkg/certificate/manager.go

```diff
@@ -1,59 +1,122 @@
 		return nil, err
 	}
 
-	client, ca, clientID, err := mrcClient.GetCertIssuerForMRC(mrcs[0])
+	err = m.init(mrcs, mrcClient)
 	if err != nil {
 		return nil, err
 	}
 
-	c := &issuer{Issuer: client, ID: clientID, CertificateAuthority: ca}
-
-	m := &Manager{
-		// The signingIssuer is responsible for signing all newly issued certificates
-		// The validatingIssuer is the issuer that issued existing certificates.
-		// its underlying cert is still in the validating trust store
-		signingIssuer:               c,
-		validatingIssuer:            c,
-		serviceCertValidityDuration: serviceCertValidityDuration,
-		msgBroker:                   msgBroker,
+	// now that the manager is initialized, start a watch
+	// we wait until the manager is initialized so that there's
+	// no lock contention between the event handler and the initial
+	// state of the manager.
+	mrcEvents, err := mrcClient.Watch(ctx)
+	if err != nil {
+		return nil, err
 	}
-	return m, nil
-}
 
-// Start takes an interval to check if the certificate
-// needs to be rotated
-func (m *Manager) Start(checkInterval time.Duration, stop <-chan struct{}) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case event, open := <-mrcEvents:
+				if !open {
+					// channel was closed; return
+					return
+				}
+				m.handleMRCEvent(event)
+			}
+		}
+	}()
+
 	ticker := time.NewTicker(checkInterval)
 	go func() {
 		m.checkAndRotate()
 		for {
 			select {
-			case <-stop:
+			case <-ctx.Done():
 				ticker.Stop()
 				return
 			case <-ticker.C:
 				m.checkAndRotate()
 			}
 		}
 	}()
+
+	return m, nil
+}
+
+// init initializes the Manager receiver based on the observed state of the cluster at this
+// instant in time. This is why we pass in a slice of MRCs; we know our Watch call will make us initially
+// consistent, we just need a valid state of the world at this instant to kick things off.
+// This method is meant to be called at initialization time, so we don't grab the lock
+func (m *Manager) init(mrcs []*v1alpha2.MeshRootCertificate, mrcClient MRCClient) error {
+	for _, mrc := range mrcs {
+		if mrc.Status.State == "errored" {
+			continue
+		}
+
+		client, ca, clientID, err := mrcClient.GetCertIssuerForMRC(mrc)
+		if err != nil {
+			return err
+		}
+
+		c := &issuer{Issuer: client, ID: clientID, CertificateAuthority: ca}
+		switch {
+		case mrc.Status.State == constants.MRCStateComplete && mrc.Status.RotationStage == constants.MRCStageIssuing:
+			m.signingIssuer = c
+			m.validatingIssuer = c
+		case mrc.Status.RotationStage == constants.MRCStageIssuing:
+			m.signingIssuer = c
+		case mrc.Status.RotationStage == constants.MRCStageValidating:
+			m.validatingIssuer = c
+		default:
+			m.signingIssuer = c
+			m.validatingIssuer = c
+		}
+	}
+
+	if m.signingIssuer == nil || m.validatingIssuer == nil {
+		return fmt.Errorf("invalid mesh root certificate configuration. Signing issuer was %#v and validating issuer was %#v", m.signingIssuer, m.validatingIssuer)
```

Comment:
nit: if logging these, we could also add a `String() string` method to issuer, to more nicely print. Not a requirement

Reply:
This error formatting is really just to show which issuer is nil. If we log the issuers elsewhere, I agree, we should add a `String()` func